addons/website_form/static/src/snippets/s_website_form/options.js

```diff
@@ -438,6 +439,30 @@ options.registry.WebsiteFormEditor = FormEditor.extend({
         }
         this._addHiddenField(value, fieldName);
     },
+    /**
+     * Prompts the user to save changes before being redirected towards an action in value
+     */
+    onClickCreateButton: function (previewMode, value, params) {
+        Dialog.confirm(this, _t("Would you like to save before being redirected? Unsaved changes will be discarded."), {
+            buttons: [
+                {
+                    text: _t("Save"),
+                    classes: 'btn-primary',
+                    close: true,
+                    click: () => {
+                        this.trigger_up('request_save').then(window.location.replace('/web#action=' + value));
```

Comment:
Ok, I am stopping my review here, this was not tested nor reviewed. `.then` expects a function to be called, you give him an expression which is probably evaluated before the trigger_up. I wondered how this cannot crash, so I tested... and it does crash and does not even save the page changes...

I am supposed to review the code when it is perfect or at the very least not crashing. If you have issues developing the feature, ask me (or anyone in the team) for help but don't "send to review and pray" :wink: